flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/action/DeleteAction.java

```diff
@@ -0,0 +1,183 @@
+        System.out.println("Action \"delete\" deletes data from a table.");
+        System.out.println();
+
+        System.out.println("Syntax:");
+        System.out.println(
+                "  delete --warehouse <warehouse-path> --database <database-name> "
+                        + "--table <table-name> [--where <filter_spec>]");
+        System.out.println("  delete --path <table-path> [--where <filter_spec>]");
+        System.out.println();
+
+        System.out.println(
+                "The '--where <filter_spec>' part is equal to the 'WHERE' clause in SQL DELETE statement");
+        System.out.println();
+
+        System.out.println("Examples:");
+        System.out.println(
+                "  delete --warehouse hdfs:///path/to/warehouse --database test_db --table test_table");
+        System.out.println("  It's equal to 'DELETE FROM test_table' which deletes all records");
+        System.out.println();
+        System.out.println(
+                "  delete --path hdfs:///path/to/warehouse/test_db.db/test_table --where id > (SELECT count(*) FROM employee)");
+        System.out.println(
+                "  It's equal to 'DELETE FROM test_table WHERE id > (SELECT count(*) FROM employee)");
+    }
+
+    @Override
+    public void run() throws Exception {
+        if (filter == null) {
+            LOG.debug("Run delete action with no filter.");
+            ((SupportsWrite) table)
+                    .deleteWhere(
+                            UUID.randomUUID().toString(),
+                            new ArrayList<>(),
+                            Lock.factory(catalog.lockFactory().orElse(null), identifier));
+        } else {
+            LOG.debug("Run delete action with filter '{}'.", filter);
+
+            StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+            env.setParallelism(1);
+            StreamTableEnvironment tEnv =
+                    StreamTableEnvironment.create(env, EnvironmentSettings.inBatchMode());
+
+            tEnv.registerCatalog(flinkCatalog.getName(), flinkCatalog);
+            tEnv.useCatalog(flinkCatalog.getName());
+
+            Table queriedTable =
+                    tEnv.sqlQuery(
+                            String.format(
+                                    "SELECT * FROM %s WHERE %s",
+                                    identifier.getEscapedFullName('`'), filter));
+
+            List<DataStructureConverter<Object, Object>> converters =
+                    queriedTable.getResolvedSchema().getColumnDataTypes().stream()
+                            .map(DataStructureConverters::getConverter)
+                            .collect(Collectors.toList());
+
+            DataStream<RowData> dataStream =
+                    tEnv.toChangelogStream(queriedTable)
+                            .map(
+                                    row -> {
+                                        BiFunction<Integer, Row, Object> fieldConverter =
```

Comment:
For performance, do use these in critical path:
- Don't use `BiFunction`.
- Don't use java stream.

Just use for loop.